sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenRefreshOptions.java

```diff
@@ -2,66 +2,89 @@
 // Licensed under the MIT License.
 package com.azure.communication.common;
 
-import java.util.function.Supplier;
 import reactor.core.publisher.Mono;
+import java.time.Duration;
+import java.util.function.Supplier;
 
 /**
  * Options for refreshing CommunicationTokenCredential
  */
 public final class CommunicationTokenRefreshOptions {
+    private static final int DEFAULT_EXPIRING_OFFSET_MINUTES = 10;
     private final Supplier<Mono<String>> tokenRefresher;
     private final boolean refreshProactively;
     private final String initialToken;
+    private final Duration refreshTimeBeforeTokenExpiry;
```

Comment:
please rename all occurrences of `refreshTimeBeforeTokenExpiry` to `refreshIntervalBeforeTokenExpiry`